src/mesh_tools.rs

```diff
@@ -150,49 +150,36 @@ pub fn separate_isolated_meshes(geometry: &Geometry) -> Vec<Geometry> {
     let mut available_face_indices: HashSet<u32> = face_to_face.keys().copied().collect();
     let mut patches: Vec<Geometry> = Vec::new();
 
-    while let Some(first_face_index) = available_face_indices.iter().next() {
-        let connected_indices = crawl_faces(*first_face_index, &face_to_face);
+    while let Some(start_face_index) = available_face_indices.iter().cloned().next() {
+        available_face_indices.remove(&start_face_index);
+        let mut index_stack = vec![start_face_index];
+        let mut connected_face_indices = HashSet::new();
```

Comment:
Let's not recreate these for each iteration, rather have just one instance of each and clean it